modules/runner/src/integrationTest/java/org/apache/ignite/internal/compute/ItLogicalTopologyTest.java

```diff
@@ -337,6 +339,27 @@ public void onNodeValidated(LogicalNode validatedNode) {
         }
     }
 
+    @Test
+    void nodeLeavesLogicalTopologyOnStop() throws Exception {
+        IgniteImpl entryNode = node(0);
+
+        IgniteImpl secondIgnite = startNode(1);
+
+        entryNode.logicalTopologyService().addEventListener(listener);
+
+        stopNode(1);
+
+        assertTrue(waitForCondition(() -> !events.isEmpty(), TimeUnit.SECONDS.toMillis(10)), "Did not see any events in time");
```

Comment:
Why do you need to use `waitForCondition` if you can simply `poll` the event queue in order to wait for an event to appear?